### PR TITLE
Update devtools for LE11

### PR DESCRIPTION
**Pull request by heitbaum**
- autoconf-archive: update to 2021.02.19
- autoconf: update to 2.71
- automake: update to 1.16.4
- bison: update to 3.7.6
- cmake: update to 3.21.2
- gcc-arm-aarch64-none-elf: update to 10.3-2021.07
- gcc-arm-aarch64-none-linux-gnu: update to 10.3-2021.07
- ~~gcc-or1k-linux: update to 2021.05-1~~
- go: update to 1.17.1
- heimdal (2 patches for build)
- kodi (Samsung): CreatePlatformDisplay with cast (update the patch)
- llvm: update to 12.0.1
- llvm: fix passing compile options building target
- m4: update to 1.4.19
- meson: update to 0.59.1
- mtools: update to 4.0.29
- qemu: remove --static to successfully compile on hirsute-proposed glibc 2.33
- qemu: update to 5.2.0
- qemu: update to 6.1.0
- rustup.rs: update to 1.24.3
- squashfs-tools: update to 4.5
- toolchain: always use cross compile and exe_wrapper with meson
- toolchain: autoconf: add workaround for >autoconf-2.69
- tools: docker: focal: update to using gcc-10
- wayland: force PKG_TOOLCHAIN=configure (was build error with `PROJECT=Samsung ARCH=arm DEVICE=Exynos`)

### CODE REVIEW

**Comment by mglae** [comment]
Commit `llvm: use array for cmake target options` is technically correct but not needed.

See `llvm: fix passing compile options building target` in #5539.

**Comment by SupervisedThinking** [comment]
Any progress at this topic? https://www.mail-archive.com/gcc-bugs@gcc.gnu.org/msg687689.html

**Comment by heitbaum** [comment]
> Commit `llvm: use array for cmake target options` is technically correct but not needed.
> 
> See `llvm: fix passing compile options building target` in #5539.

Dropped 👍

**Comment by heitbaum** [comment]
> Any progress at this topic? https://www.mail-archive.com/gcc-bugs@gcc.gnu.org/msg687689.html

I have tried a couple of times (well a 10 minute try) without success. I got all the rest updated including autoconf - which did pose a few rethinks :-) been busy hacking away at 5.14 and with my alternative boxes as of late - D1, N10c, TX6s. Think I need gcc11 to get the D1 across the line with systemd. There are a only a few reverts in my dev tree, not many … `libgpg-error, dtc, mtools, gcc, busybox` Otherwise everything else is current.

**Comment by CvH** [comment]
is this good to go ?

**Comment by mglae** [comment]
> llvm: use array for cmake target options

Again? Please include https://github.com/LibreELEC/LibreELEC.tv/pull/5539/commits/afc3e612995a33f339f4ec85abe27cc01925fc68 if this is to be merged first, I will rebase #5539.

**Comment by heitbaum** [comment]
> > llvm: use array for cmake target options
> 
> Again? Please include [afc3e61](https://github.com/LibreELEC/LibreELEC.tv/commit/afc3e612995a33f339f4ec85abe27cc01925fc68) if this is to be merged first, I will rebase #5539.

thanks for the reminder @mglae - I had forgotten whilst working with @antonlacon getting this ready to go - added and dropped the other bits. 

Build tested for:
- Generic
- RK3399

**Comment by heitbaum** [comment]
- gcc-or1k-linux: update to 2021.05-1 has been dropped (remain with 2020.08-1)

The following do not build with the `gcc-or1k-linux` update - more work to be done on that `PROJECT=Allwinner ARCH=arm DEVICE=H3 make image` and `PROJECT=Allwinner ARCH=arm DEVICE=H6 make image`

Issues are: `__stack_chk_fail` and `__stack_chk_guard`

### gcc-or1k-linux: 2021.05-1 - H3 build
```
  LD      build/scp/scp.elf
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H3.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /tmp/scp.elf.Q7fSnR.ltrans0.ltrans.o: in function `system_state_machine':
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H3.arm-11.0-devel/build/crust-0.4/common/system.c:84: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H3.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H3.arm-11.0-devel/build/crust-0.4/common/system.c:84: undefined reference to `__stack_chk_guard'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:197: build/scp/scp.elf] Error 1
```
### gcc-or1k-linux: 2021.05-1 - H6 build
```
  LD      build/scp/scp.elf
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /tmp/scp.elf.nRS4sm.ltrans0.ltrans.o: in function `axp20x_regulator_get_state':
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/regulator/axp20x.c:18: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/regulator/axp20x.c:18: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/regulator/axp20x.c:38: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/regulator/axp20x.c:38: undefined reference to `__stack_chk_fail'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /tmp/scp.elf.nRS4sm.ltrans0.ltrans.o: in function `regmap_update_bits':
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/regmap/regmap.c:67: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/regmap/regmap.c:67: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/regmap/regmap.c:76: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/regmap/regmap.c:76: undefined reference to `__stack_chk_fail'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /tmp/scp.elf.nRS4sm.ltrans0.ltrans.o: in function `axp20x_probe':
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/mfd/axp20x.c:29: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/mfd/axp20x.c:29: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/mfd/axp20x.c:49: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/drivers/mfd/axp20x.c:49: undefined reference to `__stack_chk_fail'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /tmp/scp.elf.nRS4sm.ltrans0.ltrans.o: in function `system_state_machine':
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/common/system.c:84: undefined reference to `__stack_chk_guard'
/var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/toolchain/lib/gcc-or1k-linux/bin/../lib/gcc/or1k-buildroot-linux-musl/9.4.0/../../../../or1k-buildroot-linux-musl/bin/ld: /var/media/DATA/home-rudi/LibreELEC.master/build.LibreELEC-H6.arm-11.0-devel/build/crust-0.4/common/system.c:84: undefined reference to `__stack_chk_guard'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:197: build/scp/scp.elf] Error 1
```

**Comment by heitbaum** [comment]
Successful image builds for:
```
PROJECT=Allwinner ARCH=arm DEVICE=A64 make image
PROJECT=Allwinner ARCH=arm DEVICE=H3 make image
PROJECT=Allwinner ARCH=arm DEVICE=H5 make image
PROJECT=Allwinner ARCH=arm DEVICE=H6 make image
PROJECT=Amlogic ARCH=arm DEVICE=AMLGX make image
PROJECT=Generic ARCH=x86_64 make image
PROJECT=Rockchip ARCH=arm DEVICE=RK3288 make image
PROJECT=Rockchip ARCH=arm DEVICE=RK3328 make image
PROJECT=Rockchip ARCH=arm DEVICE=RK3399 make image
PROJECT=NXP ARCH=arm DEVICE=iMX6 make image
PROJECT=NXP ARCH=arm DEVICE=iMX8 make image
PROJECT=Qualcomm ARCH=arm DEVICE=Dragonboard make image
PROJECT=RPi ARCH=arm DEVICE=RPi2 make image
PROJECT=RPi ARCH=arm DEVICE=RPi4 make image
PROJECT=Samsung ARCH=arm DEVICE=Exynos make image
  (fails in mali-midgard since kernel 5.5 / updated kodi patch included)
```
Patch for mali-midgard - https://github.com/LibreELEC/mali-midgard/pull/5

I believe this is **GtG** now.

**Review by CvH — APPROVED** [review]

File: packages/devel/glibc/package.mk [hunk 1]
```diff
@@ -45,12 +45,12 @@ post_unpack() {
...
-    aclocal --force --verbose
-    autoconf --force --verbose
-  cd -
-}
+#pre_build_target() {
+#  cd ${PKG_BUILD}
+#    aclocal --force --verbose
+#    autoconf --force --verbose
+#  cd -
+#}
```
Comment (CvH) [inline packages/devel/glibc/package.mk:53]: @heitbaum whats that ?

**Review by heitbaum — COMMENTED** [review]
Comments lines removed as suggested.

File: packages/devel/glibc/package.mk (hunk 1 above)
Comment (heitbaum) [reply packages/devel/glibc/package.mk]: Comments lines removed - as suggested.

**Comment by CvH** [comment]
@Portisch tx https://github.com/LibreELEC/LibreELEC.tv/pull/5627

**Pull request merged**